File: app/src/main/java/com/amazon/tv/leanbacklauncher/notifications/RecommendationImageKey.java
package com.amazon.tv.leanbacklauncher.notifications;

import com.amazon.tv.tvrecommendations.TvRecommendation;
import com.bumptech.glide.load.Key;

import java.security.MessageDigest;

public class RecommendationImageKey implements Key {
    private final String mKey;
    private final String mSignature;

    public RecommendationImageKey(TvRecommendation rec) {
        this.mKey = rec.getKey();
        this.mSignature = rec.getKey() + rec.getTitle() + rec.getText();
    }

    public String getKey() {
        return this.mKey;
    }

    public int hashCode() {
        return this.mSignature.hashCode();
    }

    public boolean equals(Object obj) {
        return (obj instanceof RecommendationImageKey) && ((RecommendationImageKey) obj).mSignature.equals(this.mSignature);
    }

    public void updateDiskCacheKey(MessageDigest messageDigest) {
        messageDigest.update(this.mSignature.getBytes(CHARSET));
    }
}

File: app/src/main/java/com/amazon/tv/leanbacklauncher/notifications/RecommendationImageKey.kt
package com.amazon.tv.leanbacklauncher.notifications

import com.amazon.tv.tvrecommendations.TvRecommendation
import com.bumptech.glide.load.Key
import java.security.MessageDigest

class RecommendationImageKey(rec: TvRecommendation) : Key {
    val key: String = rec.key
    private val mSignature: String
    override fun hashCode(): Int {
        return mSignature.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        return other is RecommendationImageKey && other.mSignature == mSignature
    }

    override fun updateDiskCacheKey(messageDigest: MessageDigest) {
        messageDigest.update(mSignature.toByteArray(Key.CHARSET))
    }

    init {
        mSignature = rec.key + rec.title + rec.text
    }
}